scripts/release-notes.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+MVN_VERSION_RELEASE_RAW=$1
+RELEASE_NOTES_TEMPLATE=$2
+NEWS_ITEM_TEMPLATE=$3
+BRANCH=$4
+
+# We handle both "1.2.3" and "1.2.3-M1"
+IFS='-' read -ra MVN_VERSION_RELEASE_ARRAY <<< "${MVN_VERSION_RELEASE_RAW}"
+
+export MVN_VERSION_RELEASE=${MVN_VERSION_RELEASE_ARRAY[0]}
+
+IFS='M' read -ra M_ARRAY <<< "${MVN_VERSION_RELEASE_ARRAY[1]}"
+
+# $M is the milestone build number, eg. 4 in 1.2.3-M4
+export M=${M_ARRAY[1]}
+
+if ! [[ ${BRANCH} == "" ]]; then
+  git checkout "${BRANCH}"
+  git pull
+fi
+
+echo ""
+echo "The script requires several external command line tools:"
+echo " - git"
+echo " - mvn"
+echo " - gh (the GitHub CLI, see https://github.com/cli/cli)"
+echo " - jq (https://stedolan.github.io/jq/)"
+
+echo ""
+echo "This script will stop if an unhandled error occurs";
+echo "Do not change any files in this directory while the script is running!"
+echo ""
+set -e -o pipefail
+
+if [ ! -f templates/"${RELEASE_NOTES_TEMPLATE}" ]; then
+    echo "File not found!"
+    echo "templates/${RELEASE_NOTES_TEMPLATE}"
+    exit 1;
+fi
+
+
+if [ ! -f templates/"${NEWS_ITEM_TEMPLATE}" ]; then
+    echo "File not found!"
+    echo "templates/${NEWS_ITEM_TEMPLATE}"
+    exit 1;
+fi
+
+# All builds except milestone builds need to have their milestones closed on Github before we continue.
```

Comment:
For milestones we usually tag the closed issues with a label "M1", "M2" etc. We can still do that manually if adding it to the script is difficult.